src/java.base/share/classes/jdk/internal/classfile/CodeBuilder.java

```diff
@@ -0,0 +1,1386 @@
+    }
+
+    default CodeBuilder lshl() {
+        return operatorInstruction(Opcode.LSHL);
+    }
+
+    default CodeBuilder lshr() {
+        return operatorInstruction(Opcode.LSHR);
+    }
+
+    default CodeBuilder lstore(int slot) {
+        return storeInstruction(TypeKind.LongType, slot);
+    }
+
+    default CodeBuilder lsub() {
+        return operatorInstruction(Opcode.LSUB);
+    }
+
+    default CodeBuilder lushr() {
+        return operatorInstruction(Opcode.LUSHR);
+    }
+
+    default CodeBuilder lxor() {
+        return operatorInstruction(Opcode.LXOR);
+    }
+
+    default CodeBuilder monitorenter() {
+        return monitorInstruction(Opcode.MONITORENTER);
+    }
+
+    default CodeBuilder monitorexit() {
+        return monitorInstruction(Opcode.MONITOREXIT);
+    }
+
+    default CodeBuilder multianewarray(ClassEntry array, int dims) {
+        return newMultidimensionalArrayInstruction(dims, array);
+    }
+
+    default CodeBuilder multianewarray(ClassDesc array, int dims) {
+        return newMultidimensionalArrayInstruction(dims, constantPool().classEntry(array));
+    }
+
+    default CodeBuilder new_(ClassEntry clazz) {
+        return newObjectInstruction(clazz);
+    }
+
+    default CodeBuilder new_(ClassDesc clazz) {
+        return newObjectInstruction(constantPool().classEntry(clazz));
+    }
+
+    default CodeBuilder newarray(TypeKind typeKind) {
+        return newPrimitiveArrayInstruction(typeKind);
+    }
+
+    default CodeBuilder pop() {
+        return stackInstruction(Opcode.POP);
+    }
+
+    default CodeBuilder pop2() {
+        return stackInstruction(Opcode.POP2);
+    }
+
+    default CodeBuilder putfield(FieldRefEntry ref) {
+        return fieldInstruction(Opcode.PUTFIELD, ref);
+    }
+
+    default CodeBuilder putfield(ClassDesc owner, String name, ClassDesc type) {
+        return fieldInstruction(Opcode.PUTFIELD, owner, name, type);
+    }
+
+    default CodeBuilder putstatic(FieldRefEntry ref) {
+        return fieldInstruction(Opcode.PUTSTATIC, ref);
+    }
+
+    default CodeBuilder putstatic(ClassDesc owner, String name, ClassDesc type) {
+        return fieldInstruction(Opcode.PUTSTATIC, owner, name, type);
+    }
+
+    default CodeBuilder return_() {
+        return returnInstruction(TypeKind.VoidType);
+    }
+
+    default CodeBuilder saload() {
+        return arrayLoadInstruction(TypeKind.ShortType);
+    }
+
+    default CodeBuilder sastore() {
+        return arrayStoreInstruction(TypeKind.ShortType);
+    }
+
+    default CodeBuilder sipush(int s) {
+        return constantInstruction(Opcode.SIPUSH, s);
+    }
+
+    default CodeBuilder swap() {
+        return stackInstruction(Opcode.SWAP);
+    }
+
+    default CodeBuilder tableswitch(int low, int high, Label defaultTarget, List<SwitchCase> cases) {
+        return tableSwitchInstruction(low, high, defaultTarget, cases);
+    }
+
+    default CodeBuilder tableswitch(Label defaultTarget, List<SwitchCase> cases) {
```

Comment:
`switch` seems the one instruction for which an high-level variant (like `trying`) could be useful, as generating code for that can be quite painful.

Reply:
Nice RFE, thanks.

Reply:
collected and tracked in the mailing list